lib/Share/RoomShareProvider.php

```diff
@@ -631,6 +652,35 @@ public function getSharesBy($userId, $shareType, $node, $reshares, $limit, $offs
 	 * @throws ShareNotFound
 	 */
 	public function getShareById($id, $recipientId = null): IShare {
+		if (($recipientId === null) && isset($this->sharesByIdCache[$id])) {
+			$share = $this->sharesByIdCache[$id];
+		} else {
+			$shares = $this->getSharesByIds([$id], $recipientId);
+			if (empty($shares)) {
+				throw new ShareNotFound();
+			}
+			$share = $shares[0];
+		}
+
+		if ($share === false) {
+			// Shares referring to deleted files are stored as 'false',
+			// both in the cache and in the array returned from getSharesByIds.
+			throw new ShareNotFound();
+		}
+
+		return $share;
+	}
+
+	/**
+	 * Get shares by ids
+	 *
+	 * Not part of IShareProvider API, but needed by OCA\Talk\Controller\ChatController.
+	 *
+	 * @param int[] $id
+	 * @param string|null $recipientId
+	 * @return (IShare|false)[]
```

Comment:
You don't really return false?

Reply:
Indeed! :+1: Thanks for pointing this out. Corrected by https://github.com/nextcloud/spreed/pull/7743/commits/26497b415680e2f7330c665a85142a076aab1f21 and https://github.com/nextcloud/spreed/pull/7743/commits/aa9921d461ef89d42b3da763c08747dcd99e207e.